File: Jessica/119.Pascal's Triangle2.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> res;
        for(int i=1;i<=rowIndex+1;i++){
            if(i==1 || i==2) {vector<int> temp(i,1);res.push_back(temp);continue;}
            vector<int> temp(i,1);
            for(int j=1;j<i-1;j++){
                temp[j] = res[i-2][j-1] + res[i-2][j];
            }
            
            res.push_back(temp);
        }
        return res[rowIndex];
        
        
    }
};

File: Jessica/119.Pascal's Triangle2.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0:
            return [1]
        else:
            outter=[[1]]
            for i in range(1,rowIndex+1):
                inner=[1]
                for j in range(1,i):
                    inner.append(outter[i-1][j-1]+outter[i-1][j])
                inner.append(1)
                outter.append(inner)    
        return outter[rowIndex]

